### PR TITLE
✨ Remove deprecated go get from Makefile

### PR DESCRIPTION
**Pull request by jasonliu747**
Signed-off-by: Jason Liu <jasonliu747@gmail.com>

### Ⅰ. Describe what this PR does
In Go 1.18, using go get to install go binaries is no longer supported, which breaks the go-get-tool step in the Makefile. 

This was originally taken from Kubebuilder, and it looks like the new recommended approach is to simply use go install: https://github.com/kubernetes-sigs/kubebuilder/pull/2486

This PR fixes all build dependencies to use the new method.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -157,31 +159,41 @@ deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in
...
-}
-endef
+envtest: $(ENVTEST) ## Download envtest-setup locally if necessary.
+$(ENVTEST): $(LOCALBIN)
+	GOBIN=$(LOCALBIN) go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
+
+.PHONY: golangci-lint
+golangci-lint: $(GOLANGCI_LINT) ## Download golangci-lint locally if necessary.
+$(GOLANGCI_LINT): $(LOCALBIN)
+	GOBIN=$(LOCALBIN) go install github.com/golangci/golangci-lint/cmd/golangci-lint@$(GOLANGCILINT_VERSION)
```
[the remark this answers is not in the document]
Comment (jasonliu747) [reply Makefile:199]: The latest version is 1.18, and version below 1.16 is no longer recommended to use.

[the remark this answers is not in the document]
Comment (saintube) [reply Makefile:199]: ok

**Pull request merged**